toxcore/friend_requests.c

```diff
@@ -8,18 +8,22 @@
  */
 #include "friend_requests.h"
 
-#include <stdlib.h>
 #include <string.h>
 
 #include "attributes.h"
 #include "ccompat.h"
 #include "crypto_core.h"
 #include "friend_connection.h"
+#include "mem.h"
 #include "network.h"
 #include "onion.h"
 #include "onion_announce.h"
 #include "onion_client.h"
 
+static_assert(ONION_CLIENT_MAX_DATA_SIZE <= MAX_DATA_REQUEST_SIZE, "ONION_CLIENT_MAX_DATA_SIZE is too big");
+static_assert(MAX_DATA_REQUEST_SIZE <= ONION_MAX_DATA_SIZE, "MAX_DATA_REQUEST_SIZE is too big");
+static_assert(SIZE_IPPORT <= ONION_SEND_BASE, "IP_Port does not fit in the onion packet");
```

Comment:
technically this is C32 or C++11

Reply:
https://github.com/TokTok/c-toxcore/blob/master/toxcore/ccompat.h#L66